codalab_service.py

```diff
@@ -579,9 +580,27 @@ def _run_compose_cmd(self, cmd):
         compose_env_string = ' '.join('{}={}'.format(k, v) for k, v in self.compose_env.items())
         print('(cd {}; {} {})'.format(self.compose_cwd, compose_env_string, command_string))
         if not self.args.dry_run:
-            subprocess.check_call(
-                command_string, cwd=self.compose_cwd, env=self.compose_env, shell=True
-            )
+            try:
+                popen = subprocess.Popen(
+                    command_string,
+                    cwd=self.compose_cwd,
+                    env=self.compose_env,
+                    shell=True,
+                    stdout=subprocess.PIPE,
+                    stderr=subprocess.PIPE,
+                )
+                for stdout_line in popen.stdout:
+                    print(
```

Comment:
Yuck :-)  Can we put a comment that says this will be greatly simplified with Python 3?

Reply:
Actually, this won't be simplified with python 3. The encode/decode calls are there because Travis only prints ascii. Once this is python 3, we just need to remove the "from future" import at the beginning.

Reply:
Really?  I feel like I've seen unicode being printed out on Travis.  Is it a matter of setting some encoding environment variable?